env0/resource_environment.go

```diff
@@ -23,27 +23,147 @@ func resourceEnvironment() *schema.Resource {
 			},
 			"name": {
 				Type:        schema.TypeString,
-				Description: "The environment's name",
+				Description: "the environment's name",
 				Required:    true,
 			},
 			"project_id": {
 				Type:        schema.TypeString,
 				Description: "project id of the environment",
 				Required:    true,
+				ForceNew:    true,
 			},
 			"template_id": {
 				Type:        schema.TypeString,
 				Description: "the template id the environment is to be created from",
 				Required:    true,
+				ForceNew:    true,
+			},
+			"workspace": {
+				Type:        schema.TypeString,
+				Description: "the terraform workspace of the environment",
+				Optional:    true,
+				ForceNew:    true,
+				Computed:    true,
+			},
+			"revision": {
+				Type:        schema.TypeString,
+				Description: "the revision the environment is to be run against",
+				Optional:    true,
+				Computed:    true,
+			},
+			"run_plan_on_pull_requests": {
+				Type:        schema.TypeBool,
+				Description: "should run terraform plan on pull requests creations",
+				Optional:    true,
+				Computed:    true,
+			},
+			"approve_plan_automatically": {
+				Type:        schema.TypeBool,
+				Description: "should deployments require manual approvals",
+				Optional:    true,
+				Computed:    true,
+			},
+			"deploy_on_push": {
+				Type:        schema.TypeBool,
+				Description: "should run terraform deploy on push events",
+				Optional:    true,
+				Computed:    true,
+			},
+			"auto_deploy_on_path_changes_only": {
```

Comment:
Can this attribute and `auto_deploy_by_custom_glob` co-exist?

Reply:
So can we reflect it in schema validation somehow?

Reply:
this is the only thing I found (and its not even documented on their site)
e47f7fb
it will throw this error when not both are specified (but it doesn't look like it force the field to be set to true so its still not 100%)
![image](https://user-images.githubusercontent.com/54034642/145235813-ba51b47c-1d35-4f87-b0a1-5dd132fecf3a.png)

Reply:
Yeah nothing is documented on their site, they have nice godoc structure tho
I think it should be set the other way around also right?

Reply:
thought so too but manual testing proved otherwise (its more like a dependency)